src/main/java/org/jboss/remoting3/EndpointImpl.java

```diff
@@ -686,14 +687,15 @@ private static <T> UnaryOperator<T> and(final UnaryOperator<T> first, final Unar
         return t -> second.apply(first.apply(t));
     }
 
-    public Registration addConnectionProvider(final String uriScheme, final ConnectionProviderFactory providerFactory, final OptionMap optionMap) throws IOException {
+    public Registration addConnectionProvider(final String uriScheme, final ConnectionProviderFactory providerFactory, final OptionMap optionMapIn) throws IOException {
         final SecurityManager sm = System.getSecurityManager();
         if (sm != null) {
             sm.checkPermission(RemotingPermission.ADD_CONNECTION_PROVIDER);
         }
         boolean ok = false;
         resourceUntick("Connection provider for " + uriScheme);
         try {
+            final OptionMap optionMap = EndpointBuilder.merge(optionMapIn, defaultConnectionOptionMap);
             final String saslProtocol = optionMap.get(RemotingOptions.SASL_PROTOCOL, RemotingOptions.DEFAULT_SASL_PROTOCOL);
```

Comment:
I don't want to pay the extra cost of a whole option map merge for a single get. It would rather get directly the SASL_PROTOCOL from the defaultConnectionOptionMap as a default for the optionMap

Reply:
Ahah, never mind. I see now that a few lines below we are sending the map as a parameter.